flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/filesystem/FileSystemTableSource.java

```diff
@@ -164,7 +167,7 @@ private SourceProvider createSourceProvider(BulkFormat<RowData, FileSourceSplit>
 
                     @Override
                     public TableSchema getSchema() {
-                        return schema;
+                        return TableSchema.fromResolvedSchema(schema);
```

Comment:
So migrating this isn't needed to fix the issue you had encountered?

Reply:
This code path should be hit if you're using the deprecated `FileSystemFormatFactory`, used only by csv and avro formats. As a followup I can work on migrating these 2 to the new factories, so we can prune this code completely.

Reply:
It's probably fine to leave that for now. That stack is deprecated as well anyway, so migrating it to the new Schema stack is probably not worth it.